devito/types/dimension.py

```diff
@@ -536,6 +535,81 @@ def _arg_check(self, *args, **kwargs):
 # The Dimensions below are exposed in the user API. They can only be created by
 # the user
 
+class SubDimensionThickness(DataSymbol):
+    """A DataSymbol to represent a thickness of a SubDimension"""
+
+    __rkwargs__ = DataSymbol.__rkwargs__ + ('root', 'side', 'local', 'value')
+
+    def __new__(cls, *args, **kwargs):
+        root = kwargs.pop('root')
```

Comment:
just make it explicit `__new__(cls, *args, root=None, local=None, side=None, **kwargs):